File: GCC2022/D.cpp
// import sys
// input = sys.stdin.readline

// string = input()
// arr = []

// for i in string:
//     arr.append(ord(i))

// dp = [1 for _ in range(100001)]
// check = []

// for i in range(len(arr)):
//     for j in range(i):
//         if arr[i] > arr[j] and not arr[i] in check:
//             check.append(arr[i])
//             dp[i] = max(dp[i], dp[j]+1)
//         else:
//             check = []
//             check.append(arr[i])

// print(max(dp))

#include <stdio.h>
char arr[100001];
int dp[100001];

int main(){
    for(int i=0; i<100001; i++)
}

File: GCC2022/D.py
import sys

string = str(sys.stdin.readline().rstrip())

# for i in string:
#     string.append(ord(i))

check = []
ans = 0
cnt_a = 0

for i in range(len(string)):
    if string[i] == 'A':
        # print('A발견')
        cnt_a += 1


    if string[i] == string[i-1] or ord(string[i]) == ord(string[i-1]) + 1 or i == 0:
        if string[i] == 'Z':
            # print('z발견, +',cnt_a)
            ans += cnt_a

    else:
        # print('짤림 at',i)
        cnt_a = 0
        if string[i] == 'A':
            cnt_a = 1

print(ans)